### PR TITLE
Add NonEmptyTraverse docs

### LINKED ISSUE #1747: Add NonEmptyTraverse tut documentation
As discussed [here (#1611)](https://github.com/typelevel/cats/pull/1611#pullrequestreview-40677611). Will try to work on this shortly. :)

We might want to think about docs for `Reducible` as well.

### PR DESCRIPTION
**Pull request by LukaJCB**
This should eventually resolve #1747. Still WIP however, looking for some feedback :)

### CODE REVIEW

**Comment by kailuowang** [comment]
LGTM, thanks! 👍

File: docs/src/main/tut/typeclasses/nonemptytraverse.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+layout: docs
+title:  "NonEmptyTraverse"
+section: "typeclasses"
+source: "core/src/main/scala/cats/NonEmptyTraverse.scala"
+scaladoc: "#cats.NonEmptyTraverse"
+---
+
+# NonEmptyTraverse`
+
+`NonEmptyTraverse` is a non-empty version of the `Traverse` type class, just like `Reducible` is a non-empty version of `Foldable`.
```
Comment (peterneyens) [inline docs/src/main/tut/typeclasses/nonemptytraverse.md]: Should we link to the [`Traverse`](https://typelevel.org/cats/typeclasses/traverse.html) documentation here?

Comment (LukaJCB) [reply docs/src/main/tut/typeclasses/nonemptytraverse.md]: Yes, fixed :)

File: docs/src/main/tut/typeclasses/nonemptytraverse.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+layout: docs
+title:  "NonEmptyTraverse"
+section: "typeclasses"
+source: "core/src/main/scala/cats/NonEmptyTraverse.scala"
+scaladoc: "#cats.NonEmptyTraverse"
+---
+
+# NonEmptyTraverse`
+
+`NonEmptyTraverse` is a non-empty version of the [Traverse](traverse.html) type class, just like [Reducible](reducible.html) is a non-empty version of [Foldable](foldable.html)`.
```
Comment (ceedubs) [inline docs/src/main/tut/typeclasses/nonemptytraverse.md]: I think that there's a dangling backtick on this line.

**Pull request merged**